ReversePolishNotationCalculator/MF1_ReversePolishNotationReader.cs

```diff
@@ -0,0 +1,117 @@
+namespace ReversePolishNotation;
+
+public class ReversePolishNotationReader
+{
+    public static double ReadPolishNotation(string str)
+    {
+        //if num -> save to stack
+        //if operand -> perform between two top nums
+
+        string[] instructions = str.Split();
+        Stack<double> numbers = new Stack<double>();
+
+        foreach (var op in instructions)
+        {
+            //is num or operand?
+            (InstructionType,ushort) instructionTypeTuple = GetInstructionType(op);
```

Comment:
Více čitelné by bylo použít `(InstructionType Type, ushort Length)` To lze potom používat místo Item1 a Item2. Pak jde vidět, co dané hodnoty v tuple reprezentují.